### PR TITLE
refactor(MongodbPanache): Add some tests

### PR DESCRIPTION
**Pull request by aoudiamoncef**
**Check list**:

Your pull request:

- [x] targets the `development` branch
- [x] uses the `999-SNAPSHOT` version of Quarkus
- [x] has tests (`mvn clean test`)
- [ ] works in native (`mvn clean package -Pnative`) like current development branch
- [ ] has native tests (`mvn clean verify -Pnative`) like current development branch
- [ ] makes sure the documentation must not be updated
- [ ] links the documentation update pull request (if needed)
- [ ] updates or creates the `README.md` file (with build and run instructions)
- [ ] For new quickstart, is located in the directory _component-quickstart_
- [ ] For new quickstart, is added to the root `pom.xml` and `README.md`

### CODE REVIEW

**Comment by gsmet** [comment]
@loicmathieu  this one would be for you :).

**Comment by loicmathieu** [comment]
@gsmet I have created #448 to add repository example to the quickstart. Can you have a look also so we can merge these two longstanding PRs ?

File: mongodb-panache-quickstart/src/main/java/org/acme/mongodb/panache/PersonResource.java [hunk 1]
```diff
@@ -11,20 +12,23 @@
 import javax.ws.rs.PathParam;
 import javax.ws.rs.Produces;
 import javax.ws.rs.core.Response;
+import java.util.List;
 
 @Path("/persons")
+@ApplicationScoped
```
Comment (loicmathieu) [inline mongodb-panache-quickstart/src/main/java/org/acme/mongodb/panache/PersonResource.java:18]: Not needed, annotated the class with `@Path` is enought

File: mongodb-panache-quickstart/src/test/java/org/acme/mongodb/panache/PersonResourceTest.java [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+                        .registerModule(new Jdk8Module())
+                        .registerModule(new JavaTimeModule())
+                        .disable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS)));
+    }
+
+    @Test
+    void test() {
+        PersonDTO person1 = PersonDTO.Builder
+                .aPersonDTO()
+                .withId("5889273c093d1c3e614bf2f9")
```
Comment (loicmathieu) [inline mongodb-panache-quickstart/src/test/java/org/acme/mongodb/panache/PersonResourceTest.java:55]: ID will be automatically populated by MongoDB so you can skip them

File: mongodb-panache-quickstart/src/test/java/org/acme/mongodb/panache/NativePersonResourceIT.java [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package org.acme.mongodb.panache;
+
+import io.quarkus.test.junit.NativeImageTest;
+
+@NativeImageTest
+public class NativePersonResourceIT extends PersonResourceTest {
```
Comment (loicmathieu) [inline mongodb-panache-quickstart/src/test/java/org/acme/mongodb/panache/NativePersonResourceIT.java:6]: Duplicated with `PersonResourceIT`

File: mongodb-panache-quickstart/src/main/resources/application.properties [hunk 4]
```diff
@@ -1,4 +1,6 @@
 # configure the MongoDB client for a single node cluster on localhost
 quarkus.mongodb.connection-string = mongodb://localhost:27017
 # mandatory if you don't specify the name of the database using @MongoEntity
-quarkus.mongodb.database = person
\ No newline at end of file
+quarkus.mongodb.database = persons
+
+quarkus.log.category."io.quarkus.mongodb.panache.runtime".level=DEBUG
```
Comment (loicmathieu) [inline mongodb-panache-quickstart/src/main/resources/application.properties:6]: Please avoid DEBUG level

File: mongodb-panache-quickstart/src/main/java/org/acme/mongodb/panache/Person.java [hunk 5]
```diff
@@ -1,31 +1,28 @@
...
-import java.util.List;
-
-import org.bson.codecs.pojo.annotations.BsonProperty;
-
 import io.quarkus.mongodb.panache.MongoEntity;
 import io.quarkus.mongodb.panache.PanacheMongoEntity;
+import org.bson.codecs.pojo.annotations.BsonProperty;
+
+import java.time.LocalDate;
+import java.util.List;
```
Comment (loicmathieu) [inline mongodb-panache-quickstart/src/main/java/org/acme/mongodb/panache/Person.java:8]: I' not sure about these import changes.
Are you using Quarkus code formatter ?
Did you run `mvn process-sources` ?

File: mongodb-panache-quickstart/src/main/java/org/acme/mongodb/panache/Person.java [hunk 6]
```diff
@@ -1,31 +1,28 @@
...
     public String getName() {
-        return name.toUpperCase();
+        return name.toLowerCase();
     }
 
-    // store all names in lowercase in the DB
+    // store all names in uppercase in the DB
     public void setName(String name) {
-        this.name = name.toLowerCase();
+        this.name = name.toUpperCase();
```
Comment (loicmathieu) [inline mongodb-panache-quickstart/src/main/java/org/acme/mongodb/panache/Person.java:25]: Why are you changing this ?

File: mongodb-panache-quickstart/pom.xml [hunk 7]
```diff
@@ -6,15 +6,15 @@
...
     <quarkus.platform.artifact-id>quarkus-bom</quarkus.platform.artifact-id>
     <quarkus.platform.group-id>io.quarkus</quarkus.platform.group-id>
     <quarkus.platform.version>999-SNAPSHOT</quarkus.platform.version>
-    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    <surefire-plugin.version>2.22.1</surefire-plugin.version>
+    <maven.compiler.source>1.8</maven.compiler.source>
+    <maven.compiler.target>1.8</maven.compiler.target>
     <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
   </properties>
```
Comment (loicmathieu) [inline mongodb-panache-quickstart/pom.xml:18]: Please avoid miningless changes (you just change the order of properties).

File: mongodb-panache-quickstart/src/test/java/org/acme/mongodb/panache/PersonDTO.java [hunk 8]
```diff
@@ -0,0 +1,57 @@
+package org.acme.mongodb.panache;
+
+import com.fasterxml.jackson.annotation.JsonFormat;
+
+import java.time.LocalDate;
+
+class PersonDTO {
```
Comment (loicmathieu) [inline mongodb-panache-quickstart/src/test/java/org/acme/mongodb/panache/PersonDTO.java:7]: Is it really needed ?
We want to keep the quickstart as close to the guide as possible, and we don't use such DTO in the guide.

File: mongodb-panache-quickstart/src/test/java/org/acme/mongodb/panache/PersonResourceTest.java (hunk 2 above)
Comment (loicmathieu) [reply mongodb-panache-quickstart/src/test/java/org/acme/mongodb/panache/PersonResourceTest.java:55]: The rest service should return them inside the location header so you can retrieve them from restassured response.

**Comment by aoudiamoncef** [comment]
@loicmathieu with the latest release release of testcontainers there is a built-in support for MongoDB. migrate will be easy

File: mongodb-panache-quickstart/src/main/resources/application.properties (hunk 4 above)
Comment (aoudiamoncef) [reply mongodb-panache-quickstart/src/main/resources/application.properties:6]: I could remove it ? the default value is INFO no ?

Comment (loicmathieu) [reply mongodb-panache-quickstart/src/main/resources/application.properties:6]: Yes, please remove it

File: mongodb-panache-quickstart/src/main/java/org/acme/mongodb/panache/Person.java (hunk 6 above)
Comment (aoudiamoncef) [reply mongodb-panache-quickstart/src/main/java/org/acme/mongodb/panache/Person.java:25]: Just to avoid to pass the name in capital letters in the URL.

Comment (loicmathieu) [reply mongodb-panache-quickstart/src/main/java/org/acme/mongodb/panache/Person.java:25]: OK, then

File: mongodb-panache-quickstart/src/main/java/org/acme/mongodb/panache/Person.java (hunk 5 above)
Comment (aoudiamoncef) [reply mongodb-panache-quickstart/src/main/java/org/acme/mongodb/panache/Person.java:8]: I think that it's not available on ```quarkus-quickstarts```. But i resolved this issue

File: mongodb-panache-quickstart/src/test/java/org/acme/mongodb/panache/NativePersonResourceIT.java (hunk 3 above)
Comment (aoudiamoncef) [reply mongodb-panache-quickstart/src/test/java/org/acme/mongodb/panache/NativePersonResourceIT.java:6]: I should remove which file ?

File: mongodb-panache-quickstart/src/test/java/org/acme/mongodb/panache/PersonResourceTest.java (hunk 2 above)
Comment (aoudiamoncef) [reply mongodb-panache-quickstart/src/test/java/org/acme/mongodb/panache/PersonResourceTest.java:55]: I my tests, there is nothing returned in headers by keeping the actual behaviour.
headers

**Comment by loicmathieu** [comment]
Thanks @aoudiamoncef 

@gsmet it's OK for me. It uses Testcontainers to bootstrap MongoDB, in Quarkus we use flapdoodle I don't know if we can keep Testcontianers in the quickstart.
Personally, I prefere Testcontainers than flapdoodle so I'm in favor of it.

**Comment by gsmet** [comment]
@loicmathieu thanks for the review, I merged it.

**Comment by gsmet** [comment]
Thanks @aoudiamoncef !

**Pull request merged**